tests/multi_syscall_test.rs

```diff
@@ -0,0 +1,273 @@
+    let nonce = Felt252::zero();
+
+    contract_class_cache.insert(class_hash, contract_class.clone());
+    let mut state_reader = InMemoryStateReader::default();
+    state_reader
+        .address_to_class_hash_mut()
+        .insert(address.clone(), class_hash);
+    state_reader
+        .address_to_nonce_mut()
+        .insert(address.clone(), nonce);
+
+    // Create state from the state_reader and contract cache.
+    let mut state = CachedState::new(
+        state_reader.clone(),
+        None,
+        Some(contract_class_cache.clone()),
+    );
+    // Create an execution entry point
+    let calldata = [].to_vec();
+    let caller_address = Address(0000.into());
+    let entry_point_type = EntryPointType::External;
+
+    // Block for get_caller_address.
+    {
+        let call_info = test_syscall(
+            "caller_address",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.retdata, vec![caller_address.clone().0])
+    }
+
+    // Block for get_contact_address.
+    {
+        let call_info = test_syscall(
+            "contract_address",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![])
+    }
+    // Block for get_execution_info_syscall.
+    {
+        let call_info = test_syscall(
+            "execution_info_syscall",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![]);
+    }
+
+    // Block for library_call_syscall
+    {
+        let call_info = test_syscall(
+            "replace_class_syscall_test",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![])
+    }
+
+    // Block for call_contract_syscall
+    {
+        let call_info = test_syscall(
+            "test_library_call_syscall_test",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![])
+    }
+
+    // Block for send_message_to_l1_syscall
+    {
+        let call_info = test_syscall(
+            "test_call_contract_syscall",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![])
+    }
```

Comment:
Could you check that the message you send is OK? You can use `call_info.l2_to_l1_messages`

Reply:
send_message_to_l1_syscall doesn't return anything. it returns (). So i can't compare it